monsterCatalog.py

```diff
@@ -0,0 +1,280 @@
+import flask
+import httplib2
+import json
+import requests
+import random
+import string
+import datetime
+import boto3
+import base64
+import uuid
+import werkzeug
+
+from oauth2client.client import flow_from_clientsecrets
+from oauth2client.client import FlowExchangeError
+
+import sqlalchemy
+from database import Base, User, Monster
+
+
+CLIENT_ID = json.loads(
+    open('client_secret_478230017741-rd1hon3dcgt9eoutdkjd5lsiidat2gp5.apps.googleusercontent.com.json', 'r').read())['web']['client_id']
+APPLICATION_NAME = "Monster Catalog"
+
+
+app = flask.Flask(__name__)
+
+engine = sqlalchemy.create_engine('postgres:///monstercatalog')
+Base.metadata.bind = engine
+
+DBSession = sqlalchemy.orm.sessionmaker(bind=engine)
+db_session = DBSession()
+
+
+@app.route('/')
+def main_page():
+    monsters = db_session.query(Monster).limit(20).all()
+    return flask.render_template('main.html', monsters=monsters)
+
+
+# Create anti-forgery state token
+@app.route('/signin')
+def sign_in():
+    state = ''.join(random.choice(string.ascii_uppercase + string.digits)
+                    for x in xrange(32))
+    flask.session['state'] = state
+    return flask.render_template('signup.html', STATE=state)
+
+
+@app.route('/create', methods=['GET', 'POST'])
+def create_monster():
+    # print flask.request.form
+    if 'username' not in flask.session:
+        return flask.redirect('/signin')
+    if flask.request.method == 'POST':
```

Comment:
This might be easier to read over all if you rephrase the if statement:

```python
if flask.request.method == 'GET':
  return flask.render_template('create.html')
else:
  ....
```